src/Interpreters/MySQL/InterpretersMySQLDDLQuery.cpp

```diff
@@ -92,22 +92,40 @@ static NamesAndTypesList getColumnsList(const ASTExpressionList * columns_defini
         }
 
         ASTPtr data_type = declare_column->data_type;
+        auto * data_type_function = data_type->as<ASTFunction>();
 
-        if (is_unsigned)
+        if (data_type_function)
         {
-            auto * data_type_function = data_type->as<ASTFunction>();
+            String type_name_upper = Poco::toUpper(data_type_function->name);
 
-            if (data_type_function)
+            if (is_unsigned)
             {
-                String type_name_upper = Poco::toUpper(data_type_function->name);
-
```

Comment:
Though it does not change anything, but seems like previous ordering was better.

Reply:
The new ordering allows us to share `if (data_type_function)` between the new and the old code, if `if (is_unsigned)` goes first - we will later have to have separate `if (data_type_function)` for the new code.

Reply:
Ok